HeuristicLab.Problems.DataAnalysis.Symbolic/3.4/Interpreter/NativeInterpreter.cs

```diff
@@ -80,21 +80,20 @@ public sealed class NativeInterpreter : ParameterizedNamedItem, ISymbolicDataAna
       nodes = root.IterateNodesPrefix().ToList(); nodes.Reverse();
       var code = new NativeInstruction[nodes.Count];
       if (root.SubtreeCount > ushort.MaxValue) throw new ArgumentException("Number of subtrees is too big (>65.535)");
-      int i = code.Length - 1;
-      foreach (var n in root.IterateNodesPrefix()) {
-        code[i] = new NativeInstruction { Arity = (ushort)n.SubtreeCount, OpCode = opCodeMapper(n), Length = 1, Optimize = 0 };
-        if (n is VariableTreeNode variable) {
+
+      for (int i = 0; i < code.Length; i++) {
+        code[i] = new NativeInstruction { Arity = (ushort)nodes[i].SubtreeCount, OpCode = opCodeMapper(nodes[i]), Length = 1, Optimize = 0 };
```

Comment:
we can just use `Length = nodes[i].GetLength()` and avoid a second pass below

Reply:
The suggested change is not possible as it results in evaluation errors. I have tested the changes and afterwards the unit test `TestInterpretersEstimatedValuesConsistency `fails for tree evaluations that would result in infinity, e.g. trees such as LOG(EXP(6.14537408487462 * 'x23' - EXP(7.45401367061198 * 'x12'))).

I haven't found the root cause for that, but it is definitely related to the second pass adjusting the code lengths.

Reply:
I believe this is actually an unrelated issue which occurred by coincidence while you were testing this change. The problem can be reproduced with the original code as well, if you repeat the unit test a few times. As you already noted, this seems to be an inconsistency between how C# and C++ handle extreme values. For example the expression:
`log(exp(5.4×exp(exp(5.6014)) / −4.4952))` evaluates to `−4.885 162 014 × 10^117` in C++ and to `-Infinity` in C#.